core/src/main/java/com/yahoo/oak/EntrySet.java

```diff
@@ -309,7 +292,7 @@ int getNextEntryIndex(int ei) {
         if (ei == INVALID_ENTRY_INDEX) {
             return INVALID_ENTRY_INDEX;
         }
-        return getEntryArrayFieldInt(entryIdx2intIdx(ei), OFFSET.NEXT);
+        return (int) getEntryArrayFieldLong(entryIdx2LongIdx(ei), OFFSET.NEXT);
```

Comment:
I suggest using here (and below) `getEntryArrayFieldInt()`.
At least use a mask before converting to int:
```suggestion
        return (int) getEntryArrayFieldLong(entryIdx2LongIdx(ei), OFFSET.NEXT) & ((1L << 32) - 1);
```

Reply:
Int access/assignment is currently done via long to int casting. As other part of long is not used. This is good enough.